staging-next 2022-07-29

vcunat opened the pull request:
- workflow docs: [nixos.org/manual/nixpkgs/unstable/#submitting-changes-commit-policy](https://nixos.org/manual/nixpkgs/unstable/#submitting-changes-commit-policy)
- important job(s): [hydra.nixos.org/job/nixpkgs/staging-next/unstable#tabs-constituents](https://hydra.nixos.org/job/nixpkgs/staging-next/unstable#tabs-constituents)
- jobset: [hydra.nixos.org/jobset/nixpkgs/staging-next](https://hydra.nixos.org/jobset/nixpkgs/staging-next)
- nix-review reports: https://malob.github.io/nix-review-tools-reports/
- previous staging-next: https://github.com/NixOS/nixpkgs/pull/182386

doronbehar committed:
cc-wrapper: Fortran: disable stackprotector hardening on darwin aarch64

risicle committed:
libxml2: re-enable tests for darwin

appears the lack of sandbox on most darwin machines allows the
tests to accidentally find /usr/lib/libxml2.dylib, which
causes problems

kwohlfahrt committed:
openldap: remove deprecated options

kwohlfahrt committed:
openldap: change default ldapi directory

By default, this is /run/ldapi, which is not compatible with systemd's
runtime directories. Change it to /run/slapd/ldapi (in library and
server). This makes `ldapi:///` work as a default socket again.

kwohlfahrt committed:
openldap: Allow notify outside of main thread

This fixes a bug I observed in deployment on a RPi, but not able to
reproduce in tests.

kwohlfahrt committed:
openldap: run in foreground

Now that we use notify daemon type, this works safely and simplifies
configuration.

kwohlfahrt committed:
openldap: run under systemd-defined user/group

This improves security, by starting the service as an unprivileged user,
rather than starting as root and relying on the service to drop
privileges. This requires a significant cleanup of pre-init scripts, to
make use of StateDirectory and RuntimeDirectory for permissions.

kwohlfahrt committed:
openldap: use specialisations for tests

This speeds up tests a bit. Also, ensure that mutable config works for
manual config dir.

kwohlfahrt committed:
openldap: test and fix mutable config

kwohlfahrt committed:
openldap: test starting with empty DB

This addresses the original concern behind #92544

kwohlfahrt committed:
openldap: change runtime directory

Use `openldap` for consistency between `/var/lib` and `/run`.

Artturin committed:
stdenv: substituteInPlace: accept multiple filenames

I don't know if getopt is available everywhere, so I did not use it.

in any case, it can be changed to use getopt in the future if
needed.

K900 committed:
nixos/systemd: make sure all the device nodes are created in stage1

The ConditionFileNotEmpty override patch wasn't correct for stage1, which
does have the modules in /lib. So, remove the patch and set
the right path with overrides in the final system.

Also, make sure systemd-tmpfiles-setup-dev is pulled in to create
all the necessary symlinks.

Itaros committed:
cc-wrapper: broaden explicit libc++abi linking for LLVM stdenv

risicle committed:
git: don't doInstallCheck on darwin by default

however *do* provide a `passthru.tests.withInstallCheck`.

doInstallCheck takes a ridiculous amount of time on darwin, making
staging builds ever more painful.

trofi committed:
gcc: pass --with-build-sysroot=/ for gcc builds

Without this change cross-built gcc fails to detect stack protector style:

    $ nix log -f pkgs/stdenv/linux/make-bootstrap-tools-cross.nix powerpc64le.bootGCC | fgrep __stack_chk_fail
    checking __stack_chk_fail in target C library... no
    checking __stack_chk_fail in target C library... no

It happens because gcc treats search paths differently:

    https://gcc.gnu.org/git/?p=gcc.git;a=blob;f=gcc/configure.ac;h=446747311a6aec3c810ad6aa4190f7bd383b94f7;hb=HEAD#l2458

     if test x$host != x$target || test "x$TARGET_SYSTEM_ROOT" != x ||
        test x$build != x$host || test "x$with_build_sysroot" != x; then
       ...
       if test "x$with_build_sysroot" != "x"; then
         target_header_dir="${with_build_sysroot}${native_system_header_dir}"
       elif test "x$with_sysroot" = x; then
         target_header_dir="${test_exec_prefix}/${target_noncanonical}/sys-include"
       elif test "x$with_sysroot" = xyes; then
         target_header_dir="${test_exec_prefix}/${target_noncanonical}/sys-root${native_system_header_dir}"
       else
         target_header_dir="${with_sysroot}${native_system_header_dir}"
       fi
     else
       target_header_dir=${native_system_header_dir}
     fi

By passing --with-build-sysroot=/ we trick cross-case to use
`target_header_dir="${with_sysroot}${native_system_header_dir}"`
which makes it equivalent to non-cross
`target_header_dir="${with_build_sysroot}${native_system_header_dir}"`

Tested the following setups:
- cross-compiler without libc headers (powerpc64le-static)
- cross-compiler with libc headers (powerpc64le-debug)
- cross-build compiler with libc headers (powerpc64le bootstrapTools)

Before the change only 2 of 3 compilers detected libc headers.
After the change all 3 compilers detected libc headers.

For darwin we silently ignore '-syslibroot //' argument as it does not
introduce impurities.

While at it dropped mingw special case for no-libc build. Before the change
we passed both '--without-headers --with-native-system-headers-dir' for
no-libc gcc-static builds. This tricked darwin builds to find sys/sdt.h
and fail inhibid_libc builds. Now all targets avoid passing native headers
for gcc-static builds.

While at it fixed correct headers passing to
--with-native-system-headers-dir= in host != target case: we were passing
host's headers where intention was to pass target's headers.
Noticed the mismatch as a build failure on pkgsCross.powernv.stdenv.cc
on darwin where `sys/sdt.h` is present in host's headers (libSystem)
but not target's headers (`glibc`).

Co-authored-by: Adam Joseph <54836058+amjoseph-nixpkgs@users.noreply.github.com>

trofi committed:
gcc: drop outdated sed for system headers clobber

When reviewing https://github.com/NixOS/nixpkgs/pull/181802#issuecomment-1186834504
I noticed outdated code that attempted to override /usr/include.

    sed -i \
        -e "s,glibc_header_dir=/usr/include,glibc_header_dir=$libc_dev/include", \
        gcc/configure

`glibc_header_dir` was removed from `gcc-4.6` and later in
https://gcc.gnu.org/git/?p=gcc.git;a=commitdiff;h=6961669f48aa18168b2d7daa7e2235fbec7cb636
(Dec 2010, "(gcc_cv_ld_eh_frame_hdr): Only check GNU ld for  --eh-frame-hdr.").

Since then gcc got `--with-native-system-header-dir=` which `nixpkgs` uses
for all packaged `gcc` versions.

The change should be a no-op.

mweinelt committed:
openldap: 2.6.2 -> 2.6.3

danc86 committed:
openldap: load client config from /etc, not the nix store

We want Openldap clients to load /etc/ldap.conf at runtime, not
${pkgs.openldap}/etc/ldap.conf which is always a sample config.

Pass sysconfdir=/etc at compile time, so that /etc/krb5.conf is embedded
in the library as the path of its config file.

Pass sysconfdir=${out}/etc at install time, so that the sample configs
and schema files are correctly included in the build output.

This hack works because the Makefiles are not smart enough to notice
that the sysconfdir variable has changed across invocations -- because
nobody ever writes their Makefiles to be that smart. :-)

Fixes #181937.

Commit:
qtbase: Fix build for aarch64-darwin

ajs124 committed:
glibc: remove obsolete configure option

see https://sourceware.org/bugzilla/show_bug.cgi?id=27872

ajs124 committed:
glibc: explicitly enable stack-protector

should be detected automatically

ajs124 committed:
glibc: enable Intel CET on x86

stigtsp committed:
gitMinimal: set perlSupport=false

Artturin committed:
bintools-wrapper: symlink unsymlinked binaries from -unwrapped

this shouldn't change any binary available in the default build environment
because bintools-unwrapped is already in path ( idk where it comes from but i know because objcopy is in path but not in the wrapper )

this just makes all the binaries available under 'bintools' instead of
having to use 'bintools-unwrapped'

reduces confusion because now 'objcopy' and others will be in 'bintools'

zowoq committed:
sqlite: 3.39.1 -> 3.39.2

https://www.sqlite.org/releaselog/3_39_2.html

mweinelt committed:
Merge pull request #179597 from Mic92/openldap-path

 [staging] openldap: remove deprecated options, improve encapsulation

mweinelt committed:
Merge pull request #182078 from mweinelt/openldap-2.6.3

github-actions[bot] committed:
Merge staging-next into staging

github-actions[bot] committed:
Merge staging-next into staging

flokli committed:
Merge pull request #182436 from K900/systemd-initrd-fixes

nixos/systemd: make sure all the device nodes are created in stage1

stigtsp committed:
perlPackages.LWP: 6.49 -> 6.67

Moves test dependencies to `checkInputs` to avoid pulling in
HTTP::Daemon, which is now a test requirement, if we're cross compiling
as HTTP::Daemon doesn't build with perl-cross miniperl.

github-actions[bot] committed:
Merge staging-next into staging

SuperSandro2000 committed:
libva: 2.14.0 -> 2.15.0

toonn committed:
Merge pull request #181485 from Itaros/normal-clang-llvm-redirection

cc-wrapper: broaden explicit libc++abi linking for LLVM stdenv

Artturin committed:
Merge pull request #182457 from stigtsp/package/perl-LWP-6.67

perlPackages.LWP: 6.49 -> 6.67

github-actions[bot] committed:
Merge staging-next into staging

trofi committed:
setup-hooks/strip.sh: use STRIP_FOR_TARGET, not TARGET_STRIP

Since 1ac53985 "*-wrapper; Switch from `infixSalt` to `suffixSalt`"
(2020) 'TARGET_' prefix (and infix) is no more. '_FOR_TARGET' suffix
is the only used suffix for target-specific tools and flags.

Use that in stip instead of always-empty variable.

github-actions[bot] committed:
Merge staging-next into staging

trofi committed:
setup-hooks/strip.sh: add strip{All,Debug}ListTarget variables

This change mimics existing strip{All,Debug}List variables to
allow special stripping directories just for Target.

The primary use case in mind is gcc where package has to install
both host and target ELFs. They have to be stripped by their own
strip tools accordingly.

Co-authored-by: Rick van Schijndel <Mindavi@users.noreply.github.com>
Co-authored-by: Sandro <sandro.jaeckel@gmail.com>

trofi committed:
gcc: enable stripping for cross-compilers

With explicit support for distinction between Host and Target strip paths
we can now safely strip ELF binaries with their according strip tools
without fear of damaging binaries due to architecture mismatch.

Closure size change for `pkgsCross.mingwW64.gcc12Stdenv.cc.cc`:

    # before:
    $ nix path-info -Sh $(nix-build -A pkgsCross.mingwW64.gcc12Stdenv.cc.cc) | unnix
    /<<NIX>>/x86_64-w64-mingw32-stage-final-gcc-debug-12.1.0           2.5G

    # after:
    $ nix path-info -Sh $(nix-build -A pkgsCross.mingwW64.gcc12Stdenv.cc.cc) | unnix
    /<<NIX>>/x86_64-w64-mingw32-stage-final-gcc-12.1.0         1.5G

stigtsp committed:
perlPackages.SOAPLite: add HTTPDaemon test dependency

github-actions[bot] committed:
Merge staging-next into staging

Artturin committed:
Merge pull request #182385 from Artturin/bintoolswrapper1

bintools-wrapper: symlink unsymlinked binaries from -unwrapped

lovesegfault committed:
glibc: 2.34-210 -> 2.35-163

github-actions[bot] committed:
Merge staging-next into staging

trofi committed:
setup-hooks/strip.sh: run RANLIB on static archives after stripping

'strip' does not normally preserve archive index in .a files.
This usually causes linking failures against static libs like:

    $ nix build --no-link -f. pkgsCross.mingw32.re2c
    > ...-i686-w64-mingw32-binutils-2.38/bin/i686-w64-mingw32-ld:
      /nix/store/...-i686-w64-mingw32-stage-final-gcc-13.0.0-lib/i686-w64-mingw32/lib/libstdc++.dll.a:
        error adding symbols: archive has no index; run ranlib to add one

We restore the index by running ranlib explicitly.

github-actions[bot] committed:
Merge staging-next into staging

stigtsp committed:
perlPackages.Tirex: add HTTPDaemon dependency

SCOTT-HAMILTON committed:
linux-wifi-hotspot: fix path

jtojnar committed:
python3.pkgs.pygobject3: 3.42.1 → 3.42.2

https://gitlab.gnome.org/GNOME/pygobject/-/compare/3.42.1...3.42.2

jtojnar committed:
vala: 0.56.1 → 0.56.2

https://gitlab.gnome.org/GNOME/vala/-/compare/0.56.1...0.56.2

jtojnar committed:
harfbuzz: 3.3.2 → 5.0.1

https://github.com/harfbuzz/harfbuzz/compare/3.3.2...5.0.1

jtojnar committed:
Merge pull request #182588 from jtojnar/gnome-staging

GNOME staging updates

robx committed:
openssl: split runtime dependencies of static builds into a separate output (#182444)

Artturin committed:
Merge pull request #179603 from Artturin/subplacemultiple

Ericson2314 committed:
Merge pull request #181943 from trofi/fix-cross-built-gcc

gcc: pass --with-build-sysroot=/

Mic92 committed:
go-(modules|packages): don't set trimpath for tests

trofi committed:
gcc: always enable inhibit_libc=true for --without-headers builds

It's a follow-up to the breakage caused by 21966e13d2d72b91c
("gcc: pass --with-build-sysroot=/"). It caused `pkgsLLVM`
cross-toolchain bootstrap breakage:

    $ nix build --no-link -f. pkgsLLVM.hello
    ...
    failed: /nix/store/...-x86_64-unknown-linux-gnu-stage-static-gcc-13.0.0.drv:
    ...
    configure flags: ... --enable-languages=c --disable-multilib \
      --disable-shared --enable-plugin ... --disable-libssp --disable-nls \
      --without-headers --disable-threads --disable-libgomp --disable-libquadmath \
      --disable-shared --disable-libatomic --disable-decimal-float --disable-libmpx \
      --disable-bootstrap \
      \
      --build=x86_64-unknown-linux-gnu \
      --host=x86_64-unknown-linux-gnu \
      --target=x86_64-unknown-linux-gnu
    ...
    The directory that should contain system headers does not exist:
      /usr/include
    make[2]: *** [Makefile:3279: stmp-fixinc] Error 1 shuffle=1658621302
    rm gfdl.pod gcc.pod gcov-dump.pod gcov-tool.pod fsf-funding.pod gpl.pod cpp.pod gcov.pod lto-dump.pod
    make[2]: Leaving directory '/build/build/gcc'

Note: it's a no-libc build. It's not expected to use any libc headers.
But in this case fixincludes tries to run and uses default /usr/include
location.

Fixinsludes is not normally expected to run during cross-compilation
on --without-headers. gcc/configure.ac:

    : ${inhibit_libc=false}
    if { { test x$host != x$target && test "x$with_sysroot" = x ; } ||
           test x$with_newlib = xyes ; } &&
         { test "x$with_headers" = xno || test ! -f "$target_header_dir/stdio.h"; } ; then
           inhibit_libc=true
    fi

The change explicitly passes inhibit_libc=true to configure to disable
include fixing on such cases.

Fixed `nix build --no-link -f. pkgsLLVM.hello` toolchain bootstrap.

Mic92 committed:
Merge pull request #182665 from Mic92/go-modules

go-modules: don't set trimpath for tests

github-actions[bot] committed:
Merge staging-next into staging

trofi committed:
Merge pull request #182666 from trofi/fix-non-cross-cross

gcc: always enable inhibit_libc=true for --without-headers builds

trofi committed:
Merge pull request #181994 from trofi/gcc-drop-outdated-sed

gcc: drop outdated sed for system headers clobber

trofi committed:
kmod: drop darwin support

commit 7a2c07d29bf9b "kmod: enable on darwin" added darwin
support mostly for `depmod` to be able to build `linux` kernel
on `darwin`.

In version 30 kmod requires more and more linux syscalls to build.
Instead of patching it for darwin let's drop it's support instead.

Maybe `depmod` could be packaged using some other way if needed.

Related: https://github.com/NixOS/nixpkgs/pull/182720 removed
`kmod` use on `darwin` from rare packages that still pulled it in.

trofi committed:
pciutils, ntfs3g: don't pull in `kmod` on darwin

`kmod` is a linux-only package. Previous versions worked on `darwin`
by chance and might break in the future. Remove `kmod` import from
there.

Co-authored-by: Artturi <Artturin@artturin.com>

github-actions[bot] committed:
Merge staging-next into staging

github-actions[bot] committed:
Merge staging-next into staging

github-actions[bot] committed:
Merge staging-next into staging

SuperSandro2000 committed:
Merge pull request #182469 from SuperSandro2000/libva

libva: 2.14.0 -> 2.15.0

minijackson committed:
nettle: 3.7.3 -> 3.8

github-actions[bot] committed:
Merge staging-next into staging

SuperSandro2000 committed:
Merge pull request #182720 from trofi/drop-kmod-on-darwin

pciutils, ntfs3g: don't pull in `kmod` on darwin

jtojnar committed:
Merge pull request #182616 from jtojnar/harfbuzz

harfbuzz: 3.3.2 → 5.0.1

flokli committed:
go_1_18: backport CL417615

Upstream reverted cmd/compile: revert "fix missing dict pass for type
assertions", as it's causing compilation errors in programs like gopass
- see https://github.com/gopasspw/gopass/issues/2280 and
https://github.com/NixOS/nixpkgs/issues/182440.

The revert has already been cherry-picked to HEAD, but there's no
backport release yet. This can be reverted once there's a new release
with that commit.

flokli committed:
Revert "gopass*: build with Go with CL417615"

This reverts commit 88415923b57851b292d8e74fc9a15d55f29e5a13.

With the fix being in go_1_18 itself, this gopass-specific override
doesn't need to be shipped anymore.

Artturin committed:
SDL2: restore udev support by default on linux

it was disabled seemingly by accident in 8abc1cccfbf5b72f90f71627f8a13ddccf4c3670

github-actions[bot] committed:
Merge staging-next into staging

kevinmehall committed:
opencv: Use OpenJPEG from nixpkgs instead of vendored copy

zhaofengli committed:
alsa-lib: 1.2.7.1 -> 1.2.7.2

github-actions[bot] committed:
Merge staging-next into staging

lovesegfault committed:
Merge pull request #182891 from zhaofengli/alsa-lib-1.2.7.2

alsa-lib: 1.2.7.1 -> 1.2.7.2

github-actions[bot] committed:
Merge staging-next into staging

Mic92 committed:
openal: 1.22.0 -> 1.22.2

Mic92 committed:
Merge pull request #182924 from Mic92/openal

openal: 1.22.0 -> 1.22.2

github-actions[bot] committed:
Merge staging-next into staging

onny committed:
python3Packages.tomlkit: 0.10.1 -> 0.11.1

stigtsp committed:
Merge pull request #182574 from stigtsp/package/perl-LWP-6.67-fixup

[staging] perlPackages.{SOAPLite,Tirex}: fix missing dependency

Ma27 committed:
Merge pull request #182538 from NixOS/glibc-2.35

glibc: 2.34-210 -> 2.35-163

SuperSandro2000 committed:
Merge pull request #182714 from trofi/fix-kmod-for-darwin

kmod: drop darwin support

github-actions[bot] committed:
Merge staging-next into staging

github-actions[bot] committed:
Merge staging-next into staging

github-actions[bot] committed:
Merge staging-next into staging

vcunat committed:
Merge #182810: nettle: 3.7.3 -> 3.8 (into staging)

NickCao committed:
prelink: disable tests

github-actions[bot] committed:
Merge staging-next into staging

github-actions[bot] committed:
Merge staging-next into staging

Artturin committed:
Merge pull request #183074 from NickCao/prelink-tests

prelink: disable tests

Artturin committed:
Merge pull request #182851 from Artturin/sdl2restoreudev

Artturin committed:
gobject-introspection: use objdump -p instead of prelink-rtld

Commit msg of https://github.com/openembedded/openembedded-core/commit/767e0880d4d729e659e859dd99c1cdb084b8ba51
> gobject-introspection: replace prelink-rtld with objdump -p
>
> g-i internally processes the output with regexes, and seems
> happy with what objdump is printing. It only needs to resolve
> the library name as passed to the linker to the library file name.
> Also recursive resolution (that ldd is doing and objdump is not)
> is not necessary.

Additional context section of https://github.com/NixOS/nixpkgs/issues/183071
> prelink is dropped by openembedded, support for prelink is removed from glibc,
> and itself is effectively abandoned by the upstream (yocto project).
> I think we may as well drop it rather than fix it.
> However gobject-introspection now relies on it as a runtime linker for use in cross compilation, we have to find an alternative.

risicle committed:
Merge pull request #182893 from kevinmehall/opencv-openjpeg

opencv: Use OpenJPEG from nixpkgs instead of vendored copy

Mic92 committed:
Merge pull request #182281 from helsinki-systems/feat/glibc-sec

glibc: improve configure options

Artturin committed:
libqrtr-glib: fix cross

Artturin committed:
Merge pull request #183145 from Artturin/introspeobj

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge staging-next into staging

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge staging-next into staging

lovesegfault committed:
Merge pull request #182513 from trofi/strip-for-host-and-target

gcc: enable stripping for cross-compilers

lovesegfault committed:
Merge pull request #182459 from stigtsp/fix/gitminimal-no-perl

gitMinimal: set perlSupport=false

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge staging-next into staging

r-ryantm committed:
fluidsynth: 2.2.7 -> 2.2.8

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge staging-next into staging

marsam committed:
Merge pull request #183291 from r-ryantm/auto-update/fluidsynth

fluidsynth: 2.2.7 -> 2.2.8

Artturin committed:
accountsservice: fix cross

failure line is   result = cc.run(code, name : 'value of WTMPX_FILENAME')

Artturin committed:
lightdm: fix cross

/nix/store/vkf1kli03iaxmxc9r06k01a8lmqy82k8-busybox-aarch64-unknown-linux-gnu-1.35.0/bin/which: line 1: syntax error: unterminated quoted string

r-ryantm committed:
libbpf: 0.8.0 -> 0.8.1

Artturin committed:
bintools-wrapper: symlink ar too

missed this in 1d44ac176ce6de74ac912a5b043e948a87a6d2f5

risicle committed:
Merge pull request #182967 from onny/tomlkit

python3Packages.tomlkit: Enable tests

Artturin committed:
Merge pull request #183461 from Artturin/crossfixes1

r-ryantm committed:
libwacom: 2.2.0 -> 2.4.0

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge staging-next into staging

ryantm committed:
Merge pull request #183474 from r-ryantm/auto-update/libbpf

libbpf: 0.8.0 -> 0.8.1

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge staging-next into staging

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge staging-next into staging

onny committed:
python3Packages.prompt-toolkit: Add note to failing test (#183337)

Co-authored-by: Sandro <sandro.jaeckel@gmail.com>

vcunat committed:
Merge #151983: wrapper: Fortran: disable stackprotector

...hardening on darwin aarch64 (merge into staging)

vcunat committed:
Merge #171744: libxml2: re-enable tests for darwin

...into staging

vcunat committed:
Merge #181744: git: don't doInstallCheck on darwin by default

...into staging

vcunat committed:
Merge #180327: qtbase: Fix build for aarch64-darwin

...into staging

vcunat committed:
Merge branch 'staging' into staging-next

github-actions[bot] committed:
Merge master into staging-next

lovesegfault commented:
Highlight reel:
| commit      | description |
| ------------| ----------- |
| 0e16aa7b56a | bintools-wrapper: symlink ar too |
| ce7e118975b | libbpf: 0.8.0 -> 0.8.1 |
| b21e1b5d1ab | opencv: Use OpenJPEG from nixpkgs instead of vendored copy |
| 05a57534dfa | openal: 1.22.0 -> 1.22.2 |
| 020ca659d7f | alsa-lib: 1.2.7.1 -> 1.2.7.2 |
| 5f72e1707af | nettle: 3.7.3 -> 3.8 |
| eece5d0dc0b | gcc: enable stripping for cross-compilers |
| 30a8dc71985 | libva: 2.14.0 -> 2.15.0 |
| 16e8c11629f | harfbuzz: 3.3.2 → 5.0.1 |
| 649646d7b7c | openssl: split runtime dependencies of static builds into a separate output (#182444) |
| 272fc53af16 | glibc: 2.34-210 -> 2.35-163 (glibc-2.35) |
| f348f9e3081 | sqlite: 3.39.1 -> 3.39.2 |
| e7de3e88fb3 | gitMinimal: set perlSupport=false |
| c9183d37385 | nixos/systemd: make sure all the device nodes are created in stage1 |
| b30534eb027 | openldap: load client config from /etc, not the nix store |
| e682dd84cff | bintools-wrapper: symlink unsymlinked binaries from -unwrapped |
| 1487fabf609 | glibc: enable Intel CET on x86 |
| 8f3c8aee8ad | glibc: explicitly enable stack-protector |
| 34b92568d1d | glibc: remove obsolete configure option |
| 940b020fe73 | qtbase: Fix build for aarch64-darwin |
| e11279e9629 | openldap: 2.6.2 -> 2.6.3 |
| b060076e212 | cc-wrapper: broaden explicit libc++abi linking for LLVM stdenv |
| 2cc754a7baa | cc-wrapper: Fortran: disable stackprotector hardening on darwin aarch64 |

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

jonringer committed:
dbx: update reference of path package

jonringer committed:
gnubg: use non-aliased python

jonringer committed:
nanotts: use non-aliased alsa-lib

jonringer committed:
opencpn: use non-aliased libusb1

jonringer committed:
python3Packages.jaxlib: provide non-alias default of cudnn

fabianhjr commented:
Booted into staging-next, working great. UwU

(Even a small graphical glitch on the first firefox start after booting up is gone)

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

jonringer committed:
system76-keyboard-configurator: use non-aliased pkg-config

jonringer committed:
webex: use non-aliased alsa-lib

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

K900 committed:
crystal_1_0, crystal_1_1: fix build

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

trofi committed:
nixos/openldap: fix option description markdown

Otherwise manual build fails as:

    $ nix build -f nixos/release.nix manual.x86_64-linux
    ...
     > NotImplementedError: ('md node not supported yet', 'services.openldap.declarativeContents', 'inline_html', ('<code>',))
     For full logs, run 'nix log /nix/store/pfa7bg3pflp0c31ma0rc2ahwqabs88dd-base-json-md-converted.drv'.

pennae committed:
Merge pull request #184902 from trofi/fix-openldap-markdown

nixos/openldap: fix option description markdown

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next
